query/Undefiner.java

```diff
@@ -167,14 +167,13 @@ private void undefineSub(ThingType thingType, SubConstraint subConstraint) {
             final ThingType supertype = getThingType(subConstraint.type().label().get());
             if (supertype == null) {
                 throw GraknException.of(TYPE_NOT_FOUND, subConstraint.type().label().get());
-            } else if (thingType.getSupertypes().noneMatch(t -> t.equals(supertype))
-                    && !(supertype instanceof ThingTypeImpl.Root)) {
+            } else if (thingType.getSupertypes().noneMatch(t -> t.equals(supertype))) {
```

Comment:
This line was leftover from when getSupertypes did not return the root thing type. It now does, so the callout is no longer needed.